From e773b698551d2e539698b89212022d641becfa93 Mon Sep 17 00:00:00 2001
From: Jiayang Mao <quic_jiaymao@quicinc.com>
Date: Mon, 17 Feb 2025 15:46:11 +0530
Subject: [PATCH v1] FROMLIST: arm64: dts: qcom: Add firmware-name in BT node

The qcs615-ride platform uses the QCA6698 Bluetooth chip. While the
QCA6698 shares the same IP core as the WCN6855, it has different RF
components and RAM sizes, requiring new firmware files. Use the
firmware-name property to specify the NVM and rampatch firmware to load.

Change-Id: I9c3abf2f6185b5a11c1a794ba1efde7ae845d58e
Signed-off-by: Jiayang Mao <quic_jiaymao@quicinc.com>
Upstream-Status: Submitted [https://lore.kernel.org/all/20250218074255.4152013-1-quic_jiaymao@quicinc.com/].
---
 arch/arm64/boot/dts/qcom/qcs615-ride.dts | 3 ++-
 1 file changed, 2 insertions(+), 1 deletion(-)

diff --git a/arch/arm64/boot/dts/qcom/qcs615-ride.dts b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
index 093ea5a64e76..627daed92479 100644
--- a/arch/arm64/boot/dts/qcom/qcs615-ride.dts
+++ b/arch/arm64/boot/dts/qcom/qcs615-ride.dts
@@ -1,6 +1,6 @@
 // SPDX-License-Identifier: BSD-3-Clause
 /*
- * Copyright (c) 2024, Qualcomm Innovation Center, Inc. All rights reserved.
+ * Copyright (c) 2024-2025, Qualcomm Innovation Center, Inc. All rights reserved.
  */
 /dts-v1/;
 
@@ -615,6 +615,7 @@ &uart7 {
 
 	bluetooth {
 		compatible = "qcom,wcn6855-bt";
+		firmware-name = "QCA6698/hpnv21", "QCA6698/hpbtfw21.tlv";
 
 		vddrfacmn-supply = <&vreg_pmu_rfa_cmn>;
 		vddaon-supply = <&vreg_pmu_aon_0p59>;
-- 
2.25.1

